src/threading.ts

```diff
@@ -0,0 +1,118 @@
+import interfaces = require('./interfaces');
+import IPy_Object = interfaces.IPy_Object;
+import Py_CodeObject = require('./codeobject');
+import Py_FuncObject = require('./funcobject');
+import opcodes = require('./opcodes');
+import optable = require('./optable');
+import Py_Cell = require('./cell');
+import Py_FrameObject = require('./frameobject');
+import enums = require('./enums');
+
+var maxMethodResumes: number = 10000,
+    // The number of method resumes until Doppio should yield again.
+    methodResumesLeft: number = maxMethodResumes,
+    // How responsive Ninia should aim to be, in milliseconds.
+    responsiveness: number = 1000,
+    // Used for the CMA.
+    numSamples: number = 1;
+
+class Thread{
+    // Current state of Thread
+    private status: enums.ThreadStatus = enums.ThreadStatus.NEW;
+    private stack: Py_FrameObject[] = [];
+
+    // Executes bytecode method calls
+    public run(): void {
+
+        var stack = this.stack,
+            startTime: number = (new Date()).getTime(),
+            endTime: number,
+            duration: number,
+            estMaxMethodResumes: number;
+
+        methodResumesLeft = maxMethodResumes;
+
+        if (this.status === enums.ThreadStatus.NEW) {
+           this.setStatus(enums.ThreadStatus.RUNNABLE);
```

Comment:
Actually, now that I think of it, `run` should be private, and should not setStatus to RUNNABLE. The place where you create the thread should set the thread's state to `RUNNABLE` once it is set up, which triggers the thread to run.